htdocs/core/lib/functions.lib.php

```diff
@@ -7312,9 +7313,12 @@ function make_substitutions($text, $substitutionarray, $outputlangs = null, $con
 			} else {
 				$value = dol_nl2br("$value");
 			}
-
 			$text = str_replace("$key", "$value", $text); // We must keep the " to work when value is 123.5 for example
 		}
+		if ($key == '__UNSUBSCRIBE__' && (isset($moreinheader))) {
```

Comment:
The make_substitutions() function is a technical low level method (to make a repacement of strings). We should not find code specific to one feature.

So, you must forge the string of the header
`List-Unsubscribe: <__UNSUBSCRIBE__>`
into the mailing/card.php code (because this is where you need it) and call the make_substitution on it to avec it ready to send it to the new CMailFile

Note that the mailing can also be sent using the command line script/emailings/mailing-send.php, so same action must be one inside too.

Note also that